admin/metabox/class-metabox.php

```diff
@@ -284,9 +295,7 @@ private function current_post_type_has_taxonomies() {
 	 * @return string String describing the current scope.
 	 */
 	private function determine_scope() {
-		$post_type = get_post_type( $this->get_metabox_post() );
-
-		if ( $post_type === 'page' ) {
+		if ( $this->post->post_type === 'page' ) {
```

Comment:
`$this->post` is empty when creating a new post or page, and therefore this function will always return the string `'post'`.